File: 401. Binary Watch.py
class Solution(object):

    def readBinaryWatch(self, num):

        """

        :type num: int

        :rtype: List[str]

        """

        ans=[]

        for i in range(2**10):

            if(bin(i).count('1')==num):

                minute = i & 0b0000111111

                hour = i>>6

                if hour<12 and minute<60:

                    ans.append("%d:%02d"%(hour, minute))

        return ans

File: 401. Binary Watch.cpp
class Solution {

public:

    vector<string> readBinaryWatch(int num) {

        vector<string> res;

        for(int i=0;i<1024;i++){

            if(bitset<10>(i).count()==num){

                int h = i>>6;

                int m = i & 0b0000111111;

                if(h<12 && m<60)

                    res.push_back(to_string(h) + (m < 10 ? ":0" : ":") + to_string(m));

            }

        }

        return res;

    }

};
